### ISSUE TITLE
How to view request headers?

### ISSUE THREAD

**Issue by CuriousSyrup** [mistweaverco/kulala.nvim#331]
Thanks for your work on kulala.nvim!  

I tried:  
```lua
additional_curl_options = {'--verbose'},
```  
but it didn’t show request headers. Is there a way to see them?  

Thanks!

**Comment by chunleng** [comment]
I was having the same problem and I found out you can use the following to get the headers:

```
require("kulala").setup({
	default_view = "headers_body",
})
```

ref: https://kulala.mwco.app/docs/getting-started/configuration-options#default_view

Also, this would be useful for navigating between various options:

```
require("kulala").setup({
	winbar = true,
})
```

Also, for me `opt = { winbar = true }` didn't configure properly. If it's the same for you, you might want to use the `setup` function too

**Comment by chunleng** [comment]
To code owner: I understand that the objective is to make this package simple, but maybe it's more in useful to have the default as `winbar = true` for first time user to discover this feature?

**Comment by gorillamoe** [comment]
Currently not implemented. You can use `.inspect()` when over a request to see a computed request, also with the computed request headers. Does this suffice and work for you, or do you also need additional info?

**Comment by franksl** [comment]
+1 on this, to me it's absolutely crucial to be able to view the full curl output with -v enabled, is there a way to obtain this?

**Comment by gorillamoe** [comment]
> +1 on this, to me it's absolutely crucial to be able to view the full curl output with -v enabled, is there a way to obtain this?

Not implemented as of now.

I don't feel that this would add a lot for 95% of the users of Kulala.

If this is just for one request that you need to debug further you can always copy the request as curl via Kulala and add the verbose flag manually.

**Comment by gorillamoe** [comment]
We probably have someone start working on that shortly anyway.

**Comment by YaroSpace** [comment]
Yep, I started looking into this. @gorillamoe  do you prefer  we discuss the implementation details here or elsewhere?

**Comment by gorillamoe** [comment]
> Yep, I started looking into this. [@gorillamoe](https://github.com/gorillamoe) do you prefer we discuss the implementation details here or elsewhere?

Here or on discord, but publicly.

Thanks for helping out here @YaroSpace!

**Comment by YaroSpace** [comment]
Ok, so I had a go at it.

![Image](https://github.com/user-attachments/assets/af9ef568-90dc-4f5e-bd41-5f680e3a1169)

2 questions:

1. How should we toggle the verbose mode: a) by an option in the config or b) by passing '-v/--verbose' to `additional_curl_options`?
2. In the verbose mode headers are always shown, so if the `default_view = 'headers_and_body'` headers are duplicated.  Should we just not show them in that case?

**Comment by gorillamoe** [comment]
> 1. How should we toggle the verbose mode: a) by an option in the config or b) by passing '-v/--verbose' to `additional_curl_options`?

By a config option, would be great.

> In the verbose mode headers are always shown, so if the default_view = 'headers_and_body' headers are duplicated. Should we just not show them in that case?

I think that we should introduce a new mode for that.

So maybe `default_view = 'verbose'`.

**Comment by YaroSpace** [comment]
Good idea. Verbose mode, with one set of headers.

**Comment by gorillamoe** [comment]
Released in v4.6.0.
